src/software/ai/hl/stp/tactic/crease_defender/crease_defender_fsm.cpp

```diff
@@ -193,22 +199,45 @@ std::optional<Point> CreaseDefenderFSM::findDefenseAreaIntersection(
 
 bool CreaseDefenderFSM::ballNearbyWithoutThreat(const Update& event)
 {
-    Point robot_position   = event.common.robot.position();
-    double ball_position_x = event.common.world_ptr->ball().position().x();
+    Point robot_position = event.common.robot.position();
+    Point ball_position  = event.common.world_ptr->ball().position();
+
+    std::optional<Robot> nearest_friendly =
+        event.common.world_ptr->friendlyTeam().getNearestRobot(ball_position);
     std::optional<Robot> nearest_enemy =
         event.common.world_ptr->enemyTeam().getNearestRobot(robot_position);
-    if (nearest_enemy)
+
+    if (event.control_params.ball_steal_mode == TbotsProto::BallStealMode::IGNORE)
+    {
+        // Do nothing if stealing is disabled
+        return false;
+    }
+    else if (nearest_friendly.has_value() &&
+             event.common.robot.id() != nearest_friendly.value().id())
+    {
+        // Do nothing if this robot is not the closest to the ball
+        return false;
+    }
```

Comment:
nit: Might be helpful to include that this is needed to avoid a scenario where multiple crease-defenders try to steal the ball at once.